cuqi/utilities/_utilities.py

```diff
@@ -229,3 +231,88 @@ def approx_gradient(func, x, epsilon= 0.000001):
         eps_vec[i] = 0.0
         
     return FD_gradient
+
+
+# Function for plotting 1D density functions
+def plot_1D_density(density:Density,
+                    v_min, v_max,
+                    N=500):
+    """ Plot 1D density function 
+
+    Parameters
+    ----------
+    density : CUQIpy Density
+        The density to be plotted.
+
+    v_min : float
+        Minimum value for the variable.
+    
+    v_max : float
+        Maximum value for the variable.
+
+    N : int
+        Number of grid points for the variable.
+
+    """
+    # Assert that the density is 1D
+    assert density.dim == 1, "The density must be for a scalar variable"
+    ls = np.linspace(v_min, v_max, N)
+    y = [np.exp(density.logd(grid_point)) for grid_point in ls]
+    plt.plot(ls, y)
```

Comment:
I suggest displaying the colorbar by default as well, as it might be often needed

Reply:
For 2d yes, here it is 1D

Reply:
I thought to make it more consistent with our geometry plotting method. To achieve that, now these two plotting methods accept matplotlib parameters and return matplotlib objects. Then the user can play further with it (colorbar, etc)